dEdX/src/dEdxAnalyser.cc

```diff
@@ -654,23 +827,43 @@ void dEdxAnalyser::end()
 
     for (unsigned int v=0; v<_dEdxVec[i].size(); v++)  // Normalised Resolution
     {
-      int bin = FitHist1_ampl->FindBin(_dEdxVec[i][v][0]);
-      double binMean  = FitHist1_ampl->GetBinContent(bin);
-      double normMean = _dEdxVec[i][v][1]/binMean;
-      double relError = _dEdxVec[i][v][1]/_dEdxVec[i][v][2];
+      // _dEdxVec[i][v] = mom,dEdx,track->getdEdxError(),track->getTanLambda(),(double)track->getSubdetectorHitNumbers()[_TPCindex],mom_t
+      double mom      = _dEdxVec[i][v][0];
+      double dEdx     = _dEdxVec[i][v][1];
+      int    bin      = FitHist1_mean->FindBin(mom);
+      double binMean  = FitHist1_mean->GetBinContent(bin);
+      double binError = FitHist1_mean->GetBinError(bin);
+      double normMean = dEdx/binMean;
+      double absError = sqrt( pow(_dEdxVec[i][v][2]/binMean,2) + pow(dEdx/binMean/binMean*binError,2) );
+      double relError = absError/normMean;   // _dEdxVec[i][v][1]/_dEdxVec[i][v][2]/(binError*1e8);
       double lambda   = fabs(atan(_dEdxVec[i][v][3])) *180/M_PI;
+      int    nHits    = _dEdxVec[i][v][4];
+      double mom_t    = _dEdxVec[i][v][5];
+
       if (binMean > 0)
       {
-        _ResNormHist[i]->Fill(normMean,relError);
-        if (_dEdxVec[i][v][0]>1) _ResNormHist_1GeV[i]->Fill(normMean,relError);
-        if (_dEdxVec[i][v][0]>1 && _dEdxVec[i][v][4]>=200) _NormLambdaHist[i]->Fill(lambda,normMean);
-        if (lambda<20) _NormNHitHist[i]->Fill(_dEdxVec[i][v][4],normMean);
-        _NormBothHist[i]->Fill(lambda,_dEdxVec[i][v][4],normMean);
-        if (_dEdxVec[i][v][0]>1) _NormBothHist_1GeV[i]->Fill(lambda,_dEdxVec[i][v][4],normMean);
+                                   _ResNormHist[i]       ->Fill(normMean);
+        if (mom_t>1)               _ResNormHist_1GeV[i]  ->Fill(normMean);
+        
+        if (mom_t>1 && nHits>=200) _NormLambdaHist[i]    ->Fill(lambda,normMean);
+        if (lambda<20)             _NormNHitHist[i]      ->Fill(nHits, normMean);
+        
+                     _NormBothHist[i]      ->Fill(lambda,nHits,normMean);
+        if (mom_t>1) _NormBothHist_1GeV[i] ->Fill(lambda,nHits,normMean);
+        
+        if (mom_t>1) _NormLambdaFullHist[i]     ->Fill(lambda,normMean);
+        if (mom_t>1) _NormLambdaFullHist[_nPart]->Fill(lambda,normMean);
+
+        // fiducial electrons
+        if (i==0) if (_fidMomMin<=mom && mom<=_fidMomMax && _fidLamMin<=lambda && lambda<=_fidLamMax && nHits>=_fidNHitsMin) _ResNorm_FiducialElectrons->Fill(normMean);
```

Comment:
```suggestion
        if (i==0 && _fidMomMin<=mom && mom<=_fidMomMax && _fidLamMin<=lambda && lambda<=_fidLamMax && nHits>=_fidNHitsMin) {
          _ResNorm_FiducialElectrons->Fill(normMean);
        }
```
Redundant `if` plus having everything on one line makes it very hard to parse what is actually happening if the condition is true.